modules/gateway/gateway.go

```diff
@@ -188,6 +158,16 @@ type Gateway struct {
 	mu         sync.RWMutex
 	persistDir string
 	threads    siasync.ThreadGroup
+
+	// Unique ID
+	id gatewayID
+}
+
+type gatewayID [8]byte
+
+func randGatewayID() (id gatewayID, err error) {
+	_, err = rand.Read(id[:])
+	return
```

Comment:
Do you just want to do `err = rand.Read(g.id[:])` directly in the `New` function then?